src/blockmap.jl

```diff
@@ -445,17 +447,12 @@ for k in 1:8 # is 8 sufficient?
     mapargs = ntuple(n -> :(LinearMap($(Symbol(:A,n)))), Val(k-1))
     # yields (:LinearMap(A1), :LinearMap(A2), ..., :LinearMap(A(k-1)))
 
-    @eval begin
-        function SparseArrays.blockdiag($(Is...), $L, As::Union{LinearMap,AbstractMatrix}...)
+    # since the below method is more specific than the Base method, it would redefine Base behavior
+    @eval function Base.cat($(Is...), $L, As::MapOrMatrix...; dims::Dims{2})
```

Comment:
The rationale above does not work here: `cat` is defined for generic arguments, so our method would be more specific and hence take over matrix concatenation, for instance. That's why we need to keep this requirement that among the first 8 arguments there is a `LinearMap`.

Reply:
Hm, I just noticed that it doesn't apply to `blockdiag` either. We'd be overloading `SparseArrays.blockdiag` for `AbstractMatrix`s, i.e., not our own type. 🏴‍☠️